File: ClinicControlCenter/Domain/ViewModels/AppointmentViewModel.cs
using System;

namespace ClinicControlCenter.Domain.ViewModels {
    public class AppointmentViewModel {
        public DateTimeOffset Date { get; set; }

        public string Name { get; set; }

        public string Email { get; set; }

        public string Telephone { get; set; }

        public string DoctorId { get; set; } //FK para Doctor

        public virtual DoctorViewModel Doctor { get; set; }
    }
}

File: ClinicControlCenter/ClientApp/src/libs/domain/models/appointment/AppointmentViewModel.ts
import { IDoctorViewModel, IUserViewModel } from "src/app/system-management/models/UserViewModel.model";

export interface IAppointmentViewModel {
  Date: string;
  Name: string;
  Email: string;
  Telephone: string;
  DoctorId: string;
  User?: IUserViewModel;
  Doctor?: IDoctorViewModel;
}
